src/main/java/com/github/manolo8/darkbot/gui/tree/editors/FileEditor.java

```diff
@@ -0,0 +1,83 @@
+package com.github.manolo8.darkbot.gui.tree.editors;
+
+import com.github.manolo8.darkbot.gui.AdvancedConfig;
+import com.google.gson.*;
+import eu.darkbot.api.config.ConfigSetting;
+import eu.darkbot.api.config.util.OptionEditor;
+
+import javax.swing.*;
+import java.awt.*;
+import java.io.File;
+import java.lang.reflect.Type;
+import java.util.function.Consumer;
+
+public class FileEditor extends JButton implements OptionEditor<File> {
+    private JFileChooser fc;
+    private File file;
+    private final Consumer<JTree> stopEditing = JTree::stopEditing;
+
+    public FileEditor() {
+        setFocusable(false);
+        addActionListener(e -> stopEditing.accept((JTree) SwingUtilities.getAncestorOfClass(JTree.class, FileEditor.this)));
```

Comment:
These 2 are very questionable choices
I believe the button needs to remain focusable, at the very least for keyboard support, but also because otherwise i'm not sure what will even be focused by the tree node when editing
The second, assumes a tree is available, which isn't necessarily always available, also, when the button is pressed the pop-up to pick a file should be opened, not edition stopped, so it just feels pretty counter intuitive as to what's really going on here

Reply:
You right, it not work with keyboard(
I added this trick because new select files window triggered of releasing focus from button so as result not displayed new file name however saves to tree. It not reproduce if button setFocusable(false). 
This trick make possible to show new file name. I can't found another way to fix it. I will try to do it tomorrow.

Reply:
How does this work on the image picker? does it work there?
in theory when saving, either the tree editor should still be triggered, in which case, the name should just change by  your call to "setEditing" (which calls setText), if it's no longer focused, it's weird that it would even properly work to save the setting at all, as you need to reply to getEditorValue with the new value for it to be updated